### PR TITLE
[Tests] NFC: Add few associated type inference test-cases

### PR DESCRIPTION
**Pull request by xedin**

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test macOS platform

**Comment by xedin** [comment]
@swift-ci please test Windows platform

**Comment by xedin** [comment]
@swift-ci please test

File: test/decl/protocol/req/rdar123357062.swift
Comment (AnthonyLatsis) [inline test/decl/protocol/req/rdar123357062.swift:1]: This one looks like it may fit better in the validation test suite. Just a thought.

**Review by AnthonyLatsis — APPROVED** [review]

File: test/decl/protocol/req/rdar123357062.swift
Comment (xedin) [reply test/decl/protocol/req/rdar123357062.swift:1]: I'll keep it here for now, @slavapestov has plans to some of this structure.

**Pull request merged**